src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs

```diff
@@ -190,6 +190,41 @@ public void CleanAllErrors()
             }
         }
 
+        /// <summary>
+        /// Remove the specified error from the Error List.
+        /// </summary>
+        /// <param name="errorToRemove">
+        /// The error to remove from the Error List.
+        /// </param>
+        public void RemoveError(SarifErrorListItem errorToRemove)
+        {
+            using (this.tableEntriesLock.EnterWriteLock())
+            {
+                // Find the dictionary entry for the one log file that contains the error to be
+                // removed. Any given error object can appear in at most one log file, even if
+                // multiple log files report the "same" error.
+                List<SarifResultTableEntry> tableEntryListWithSpecifiedError =
+                    this.logFileToTableEntries.Values
+                    .SingleOrDefault(tableEntryList => tableEntryList.Select(tableEntry => tableEntry.Error).Contains(errorToRemove));
+
+                if (tableEntryListWithSpecifiedError != null)
+                {
+                    // Any give error object can appear at most once in any log file, even if the
+                    // log file reports the "same" error multiple times. And we've already seen
+                    // that this error object does appear in this log file, so calling Single is
+                    // just fine.
+                    SarifResultTableEntry entryToRemove = tableEntryListWithSpecifiedError.Single(tableEntry => tableEntry.Error == errorToRemove);
+
+                    this.CallSinks(
+                        sink => sink.RemoveEntries(new List<SarifResultTableEntry> { entryToRemove }));
```

Comment:
>new List<SarifResultTableEntry> { entryToRemove }) [](start = 51, length = 50)

surprised new [] { entryToRemove } isn't sufficient :)

Reply:
I see this has been merged, so I'll do it in the next PR.

---
In reply to: [516775691](https://github.com/microsoft/sarif-visualstudio-extension/pull/249#discussion_r516775691) [](ancestors = 516775691)